--- a/agents/gce/fence_gce.py	2021-10-19 10:29:17.000000000 +0000
+++ b/agents/gce/fence_gce.py	2021-10-26 15:26:11.348781027 +0000
@@ -1,7 +1,6 @@
 #!@PYTHON@ -tt
 
 import atexit
-import httplib2
 import logging
 import os
 import sys
@@ -19,6 +18,7 @@
 from fencing import fail_usage, run_delay, all_opt, atexit_handler, check_input, process_input, show_docs, fence_action
 try:
   sys.path.insert(0, '/usr/lib/fence-agents/bundled/google')
+  import httplib2
   import googleapiclient.discovery
   import socks
   try:
