--- a/agents/scsi/fence_scsi.py 2024-01-03 14:15:20.755284113 +0100
+++ b/agents/scsi/fence_scsi.py 2024-01-03 12:32:01.598598127 +0100
@@ -190,7 +190,8 @@
 	cmd = options["--sg_persist-path"] + " -n -i " + opts + "-r -d " + dev
 	out = run_cmd(options, cmd)
 	if out["rc"] and fail:
-		fail_usage("Cannot get reservation key")
+		fail_usage('Cannot get reservation key on device "' + dev
+                        + '": ' + out["err"])
 	match = re.search(r"\s+key=0x(\S+)\s+", out["out"], re.IGNORECASE)
 	return match.group(1) if match else None
 
@@ -204,7 +205,8 @@
 	cmd = options["--sg_persist-path"] + " -n -i " + opts + "-k -d " + dev
 	out = run_cmd(options, cmd)
 	if out["rc"]:
-		fail_usage("Cannot get registration keys", fail)
+		fail_usage('Cannot get registration keys on device "' + dev
+                        + '": ' + out["err"], fail)
 		if not fail:
 			return []
 	for line in out["out"].split("\n"):
