--- a/google/pkg_resources/__init__.py	2025-06-12 09:41:54.535219946 +0000
+++ b/google/pkg_resources/__init__.py	2025-06-12 09:42:07.508276281 +0000
@@ -95,16 +95,6 @@
     from _typeshed.importlib import LoaderProtocol
     from typing_extensions import Self, TypeAlias
 
-warnings.warn(
-    "pkg_resources is deprecated as an API. "
-    "See https://setuptools.pypa.io/en/latest/pkg_resources.html. "
-    "The pkg_resources package is slated for removal as early as "
-    "2025-11-30. Refrain from using this package or pin to "
-    "Setuptools<81.",
-    UserWarning,
-    stacklevel=2,
-)
-
 _T = TypeVar("_T")
 _DistributionT = TypeVar("_DistributionT", bound="Distribution")
 # Type aliases
--- a/kubevirt/pkg_resources/__init__.py	2025-06-12 09:41:54.535219946 +0000
+++ b/kubevirt/pkg_resources/__init__.py	2025-06-12 09:42:07.508276281 +0000
@@ -95,16 +95,6 @@
     from _typeshed.importlib import LoaderProtocol
     from typing_extensions import Self, TypeAlias
 
-warnings.warn(
-    "pkg_resources is deprecated as an API. "
-    "See https://setuptools.pypa.io/en/latest/pkg_resources.html. "
-    "The pkg_resources package is slated for removal as early as "
-    "2025-11-30. Refrain from using this package or pin to "
-    "Setuptools<81.",
-    UserWarning,
-    stacklevel=2,
-)
-
 _T = TypeVar("_T")
 _DistributionT = TypeVar("_DistributionT", bound="Distribution")
 # Type aliases
